By default, cryptography tries to load the OpenSSL legacy provider and
emits a warning when it's not available.

The warning can be suppressed by setting the CRYPTOGRAPHY_OPENSSL_NO_LEGACY
environment variable, but since Solaris considers the legacy provider
an extra package that should only be installed when really necessary,
this patch removes the warning.

Several related upstream issues for reference:
https://github.com/pyca/cryptography/issues/7358
https://github.com/pyca/cryptography/pull/7650
https://github.com/pyca/cryptography/pull/12601

--- cryptography-46.0.3/src/rust/src/lib.rs
+++ cryptography-46.0.3/src/rust/src/lib.rs
@@ -68,10 +68,6 @@ fn _initialize_providers(py: pyo3::Pytho
     let legacy = if load_legacy {
         let legacy_result = provider::Provider::load(None, "legacy");
         if legacy_result.is_err() {
-            let message = crate::utils::cstr_from_literal!("OpenSSL 3's legacy provider failed to load. Legacy algorithms will not be available. If you need those algorithms, check your OpenSSL configuration.");
-            let warning_cls = pyo3::exceptions::PyWarning::type_object(py).into_any();
-            pyo3::PyErr::warn(py, &warning_cls, message, 1)?;
-
             None
         } else {
             Some(legacy_result?)
